server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/MaxRetryAllocationDecider.java

```diff
@@ -73,4 +73,9 @@ public Decision canForceAllocatePrimary(ShardRouting shardRouting, RoutingNode n
         // if so, we don't want to force the primary allocation here
         return canAllocate(shardRouting, node, allocation);
     }
+
+    @Override
+    public Decision canForceDuringVacate(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
+        return canAllocate(shardRouting, node, allocation);
```

Comment:
I wonder if we want a higher limit than 5 retries here?

Reply:
Something like 10 retries maybe? Double it?

Reply:
This was primarily relevant in the scheme where we tried against all other nodes in the cluster. Given newest developments, I think we can leave this as is (i.e., 5 retries).